lib/translator/inquiry.cc

```diff
@@ -187,6 +187,109 @@ void TranslateUnitSerialNumberVpd(
   memcpy(&buffer[sizeof(result)], &product_serial_number, result.page_length);
 }
 
+void TranslateLogicalBlockProvisioningVpd(
+    const nvme_defs::IdentifyControllerData &identify_controller_data,
+    const nvme_defs::IdentifyNamespace &identify_namespace_data,
+    absl::Span<uint8_t> buffer) {
+  bool ad = identify_controller_data.oncs.dsm;
+
+  scsi_defs::LogicalBlockProvisioningVpd result = {
+      .page_code = scsi_defs::PageCode::kBlockLimitsVpd,
+      .page_length = 0x04,
+
+      /*
+      THRESHOLD_EXPONENT
+      Shall be set to 00h to indicate that there are no thin
+      provisioning thresholds
+      This would require modification if thin-provisioning is
+      supported
+
+      TODO:
+      Right now we assume this field is 0.
+      If thin provisioning is enabled, we can calculate a valid threshold
+      exponnet from the READ CAPACITY (16) parameter data.
+      */
+
+      // .threshold_exponent= identify_namespace_data.nsfeat.thin_prov ?
+      // read_capacity : 0,
+
+      /*
+      LBPRZ
+      Shall be set to 1 if Dataset Management command –
+      Deallocate (AD) attribute is supported and the device
+      returns all zeros for reads of deallocated LBAs, otherwise
+      set to 0.
+
+      TODO: the AD attribute itself may not guarantee the device
+      returns all zeros for reads of deallocated LBAs.
+      May have to double check this one if there are any errors.
+      */
```

Comment:
nit: Google C++ style guide recommends sticking to one style of making comments on the file:

https://google.github.io/styleguide/cppguide.html#Comment_Style